modules/ui/org.eclipse.fx.ui.controls/src/main/java/org/eclipse/fx/ui/controls/table/TableViewUtil.java

```diff
@@ -180,24 +217,38 @@
 	public static <S> TableView<S> captureState(TableView<S> view, Memento m, Function<TableColumn<S, ?>, String> columnKeyProvider, TableView_Properties... properties) {
 		Map<TableColumn<S, ?>, String> nodeToIdMap = view.getColumns().stream().collect(Collectors.toMap(c -> c, columnKeyProvider));
 
-		for (TableColumn<S, ?> c : view.getColumns()) {
-			String key = nodeToIdMap.get(c);
-			if (key == null) {
-				continue;
+		view.getColumns().addListener((ListChangeListener<TableColumn<S, ?>>) change -> {
+			change.reset();

```

Comment:
why is reset needed here?

Reply:
workaround for jdk bug 8164096 (which is fixed in u122)